tel2puml/find_unique_graphs/otel_ingestion/otel_data_source.py

```diff
@@ -1,16 +1,26 @@
 """Module containing classes to handle different OTel data sources."""
 
+import os
+import yaml
+import time
+import ijson
 from abc import ABC, abstractmethod
-from typing import Self
+from typing import Self, Any, Iterator
 
 from tel2puml.find_unique_graphs.otel_ingestion.otel_data_model import (
     OTelEvent,
+    JSONDataSourceConfig,
 )
+from tel2puml.find_unique_graphs.otel_ingestion import json_data_converter
 
 
 class OTELDataSource(ABC):
     """Abstract class for returning a OTelEvent object from a data source."""
 
+    def __init__(self) -> None:
+        """Constructor method."""
+        self.valid_data_sources = ["json"]
```

Comment:
I don't think we'll need this as we'll be selecting JSONDataSource based on the config. If we want to keep a list anywhere of mappings say we could keep it at the class level on OTELDataSource